graphqlapi/src/main/java/io/stargate/graphql/GraphqlActivator.java

```diff
@@ -87,6 +96,19 @@ private synchronized void maybeStopService() {
       } catch (Exception e) {
         LOG.error("Unexpected error while stopping GraphQL", e);
       }
+      graphqlHealthCheck.healthy = false;
+    }
+  }
+
+  private static class GraphqlHealthCheck extends HealthCheck {
+
+    private volatile boolean healthy = false;
+
+    @Override
+    protected Result check() throws Exception {
```

Comment:
This check looks good.
For the Kafka CDC producer I would like to expose a list of checks:
`ZookeperHealthCheck`, `KafkaHealthCheck` and `SchemaRegistryHealthCheck`.
With the current solution, I can construct one result:
```
Result.builder().
              withMessage("Kafka cdc Producer running")
              .withDetail("kafka", "UP")
              .withDetail("zookeeper", "UP")
              .withDetail("schema-registry", "UP")
              .build()
```
However, I wonder that maybe it will be better to register N health-checks for every system:
```
Result kafkaHealthCheck = Result.healthy("kafka UP");
Result zookeeperHealthCheck = Result.healthy("zookeeper UP");
healthCheckRegistry.get().register("kafkaCdcProducer.kafka", kafkaHealthCheck);
healthCheckRegistry.get().register("kafkaCdcProducer.zookeeper", zookeeperHealthCheck);
```
What is important, both solutions seem feasible using the approach implemented here so I am +1 on it.

Reply:
Yeah the keys are simple strings so there are no constraints on the client bundles. Registering multiple checks for a given API sounds like a perfectly valid approach.